### ISSUE TITLE
The Mod Field

### ISSUE THREAD

**Issue by blond** [bem-sdk/bem-naming#95]
I think we should use the `mod` name instead of `modName` with `modVal`.

Now:

``` js
{ block: 'button', modName: 'size', modVal: 's' },
{ block: 'button', modName: 'disabled', modVal: true }
```

Need:

``` js
{ block: 'button', mod: { name: 'size', val: 's' } },
{ block: 'button', mod: 'disabled' }
```

**Why?**
1. The record will be easier for boolean modifiers.
2. There will be less confusion with the `{ block: 'button', modName: 'size', modVal: false }`.
3. The modifier is always described by one field.
4. It will be easier to define the type of entity.

**Comment by apsavin** [comment]
I did not get 2.
`{ block: 'button', mod: { name: 'size', val: false } }` 
vs
 `{ block: 'button', modName: 'size', modVal: false }`
Why less confusion?

**Comment by blond** [comment]
@apsavin I think that now `{ block: 'button', modName: 'size', modVal: false }` will consider equivalent `{ block: 'button', modName: 'size' }`.

If for boolean modifiers will the short form `{ block: 'button', mod: 'disabled' }` then no one will want to write `{ block: 'button', mod: { name: 'disabled', value: true }` or `{ block: 'button', mod: { name: 'disabled', value: false }` and the full form will always be used for `key-value` modifiers.

**Comment by vithar** [comment]
👍

**Comment by tadatuta** [comment]
@blond isn't it just yet one more format for absolutely same purpose?

are there **real** places one must write such form manually (except for `i-bem` which you won't change anyways)?

**Comment by blond** [comment]
@tadatuta `i-bem` works with BEM entities, but `bem-naming` works with BEM entity parts. It's means that the syntax does not necessarily coincide.

For example, `i-bem` can use `mods: { key: 'value' }`. 

In addition the `i-bem` and `bem-naming` have different users. It's means that the consistency of the formats is not important in this case.

In my opinion the `mod` field is better than `modName` + `modVal` fields. So I still want to make `mod` field.

**Comment by blond** [comment]
@tadatuta also `i-bem` can support `mod` field with old `modName + modVal` option

**Comment by tadatuta** [comment]
> also i-bem can support mod field with old modName + modVal option

?

**Comment by blond** [comment]
[bem-entity-name](https://github.com/bem-sdk/bem-entity-name) use `mod` field and support `modName` and `modVal` for backward compatibility.

I think `bem-naming` must do the same.

**Comment by blond** [comment]
Fixed in #148
